src/sparseml/optim/manager.py

```diff
@@ -93,6 +103,7 @@ def compose_staged(
         cls,
         base_recipe: Union[str, "BaseManager"],
         additional_recipe: Union[str, "BaseManager"],
+        additional_metadata=Dict[str, Any],
```

Comment:
Being pretty stubborn about pushing metadata merging dictionary here, let's discuss it.